### ISSUE TITLE
Exception on task preparation

### ISSUE THREAD

**Issue by toddams** [synhershko/NAppUpdate#72]
I'm getting "task prepare failed" exception while calling UpdateManager.Instance.PrepareUpdates. 
Now I can't figure out what the actual problem. I tried to look through Exception's Stack Trace, but it's almost empty. Here is a source code:

```
try
{
    task.Prepare(UpdateSource);
}
catch (Exception ex)
{
    task.ExecutionStatus = TaskExecutionStatus.FailedToPrepare;
    Logger.Log(ex);
    throw new UpdateProcessFailedException("Failed to prepare task: " + task.Description, ex);
}
```
If exception occurs while preparing a task - it is re-trown, so exception stack is empty and you can't find out a real problem reason.

My suggestion is to include it in inner exception, so you can see it's stack trace

**Comment by synhershko** [comment]
The original exception is included, as you can see in the code you included. What is missing?

**Comment by toddams** [comment]
Del. My bad, there is already innerEx

**Comment by toddams** [comment]
I found out a reason of exception, and want to share:

Code was like this
```
manager.CheckForUpdates(updateSource);
manager.PrepareUpdates();
```
And I got NullReferenceException on PrepareUpdates(). Then I set UpdateSource property instead of passing it like a method parameter, like this:
```
var manager = UpdateManager.Instance;
manager.UpdateSource = updateSource;
manager.CheckForUpdates();
manager.PrepareUpdates();
```
And it works.

**Comment by robinwassen** [comment]
I would actually consider this as a design bug and recommend that the `CheckForUpdates(IUpdateSource source)` method to either be removed or changed so it sets the UpdateManager.UpdateSource property when called.

The reason is that it is strange to offer two alternatives pass in the update source when performing updates, but only one works for the whole process.

**Comment by robinwassen** [comment]
Thank you for reporting this @toddams !

**Comment by robinwassen** [comment]
This has been resolved in #73
